internal/code/compare_test.go

```diff
@@ -63,6 +63,25 @@ func TestCompatibleTypes(t *testing.T) {
 	}
 }
 
+func TestSimilarBasicKind(t *testing.T) {
+	valid := []struct {
+		name     string
+		expected types.BasicKind
+		actual   types.BasicKind
+	}{
+		{name: "string=string", expected: types.String, actual: types.String},
+		{name: "int8=int", expected: types.Int64, actual: types.Int8},
+		{name: "byte=byte", expected: types.Byte, actual: types.Byte},
+	}
+
+	for _, tc := range valid {
+		t.Run(tc.name, func(t *testing.T) {
+			require.Equal(t, tc.expected, similarBasicKind(tc.actual))
+
+		})
+	}
+}
+
```

Comment:
I upgraded your test with the following to include all `BasicKind` defined by [Go](https://github.com/golang/go/blob/a11b8e37652ff60302b4a4a55a8a43db6a066ecf/src/go/types/basic.go#L10), and because `Byte` is an alias to `Uint8`, your code in `similarBasicKind()` is failing in test `Uint8=Int64`. I am trying to find a way to make it work.

```Go
func TestSimilarBasicKind(t *testing.T) {
	valid := []struct {
		name     string
		expected types.BasicKind
		actual   types.BasicKind
	}{
		// invalid
		{name: "Invalid=Invalid", expected: types.Invalid, actual: types.Invalid},

		// predeclared types
		{name: "Bool=Bool", expected: types.Bool, actual: types.Bool},
		{name: "Int=Int", expected: types.Int, actual: types.Int},
		{name: "Int8=Int64", expected: types.Int64, actual: types.Int8},
		{name: "Int16=Int64", expected: types.Int64, actual: types.Int16},
		{name: "Int32=Int32", expected: types.Int32, actual: types.Int32},
		{name: "Int64=Int64", expected: types.Int64, actual: types.Int64},
		{name: "Uint=Int64", expected: types.Int64, actual: types.Uint},
		{name: "Uint8=Int64", expected: types.Int64, actual: types.Uint8},
		{name: "Uint16=Int64", expected: types.Int64, actual: types.Uint16},
		{name: "Uint32=Int64", expected: types.Int64, actual: types.Uint32},
		{name: "Uint64=Int64", expected: types.Uint64, actual: types.Uint64},
		{name: "Uintptr=Uintptr", expected: types.Uintptr, actual: types.Uintptr},
		{name: "Float32=Float32", expected: types.Float32, actual: types.Float32},
		{name: "Float64=Float64", expected: types.Float64, actual: types.Float64},
		{name: "Complex64=Complex64", expected: types.Complex64, actual: types.Complex64},
		{name: "Complex128=Complex128", expected: types.Complex128, actual: types.Complex128},
		{name: "String=String", expected: types.String, actual: types.String},
		{name: "UnsafePointer=UnsafePointer", expected: types.UnsafePointer, actual: types.UnsafePointer},

		// types for untyped values
		{name: "UntypedBool=UntypedBool", expected: types.UntypedBool, actual: types.UntypedBool},
		{name: "UntypedInt=UntypedInt", expected: types.UntypedInt, actual: types.UntypedInt},
		{name: "UntypedRune=UntypedRune", expected: types.UntypedRune, actual: types.UntypedRune},
		{name: "UntypedFloat=UntypedFloat", expected: types.UntypedFloat, actual: types.UntypedFloat},
		{name: "UntypedComplex=UntypedComplex", expected: types.UntypedComplex, actual: types.UntypedComplex},
		{name: "UntypedString=UntypedString", expected: types.UntypedString, actual: types.UntypedString},
		{name: "UntypedNil=UntypedNil", expected: types.UntypedNil, actual: types.UntypedNil},

		// aliases
		{name: "Byte=Byte", expected: types.Byte, actual: types.Byte},
		{name: "Rune=Rune", expected: types.Rune, actual: types.Rune},
	}

	for _, tc := range valid {
		t.Run(tc.name, func(t *testing.T) {
			require.Equal(t, tc.expected, similarBasicKind(tc.actual))
		})
	}
}
```

Reply:
Thanks for the reply, yes that would be perfect to check them all, but I think that my suggestion to use this check in the comparison before the method:

```go
if actualBasic.Kind() != expected.Kind() &&  similarBasicKind(actualBasic.Kind()) != expected.Kind() {
		return fmt.Errorf("basic kind differs, %s != %s", expected.Name(), actualBasic.Name())
	}
```

would be simpler and wouldn't require the test.

Reply:
Yes, I am testing your suggestion by adding some tests in `_examples/scalars`. Can you provide me with your `MarshalBytes/UnmarshalBytes` functions?

Reply:
Sorry, just found out about this [file](https://github.com/99designs/gqlgen/blob/master/codegen/testserver/singlefile/bytes.go), so I will use it for my tests.